core/src/main/scala/org/apache/spark/ui/storage/StorageTab.scala

```diff
@@ -39,7 +39,9 @@ private[ui] class StorageTab(parent: SparkUI) extends SparkUITab(parent, "storag
  * This class is thread-safe (unlike JobProgressListener)
  */
 @DeveloperApi
-class StorageListener(storageStatusListener: StorageStatusListener) extends SparkListener {
+class StorageListener(storageStatusListener: StorageStatusListener)
+  extends BlockStatusListener with SparkListener {
```

Comment:
AFAIK you can just extend `BlockStatusListener` here, which will transitively pull in the `SparkListener` interface.